packages/eas-cli/src/commands/env/pull.ts

```diff
@@ -0,0 +1,109 @@
+import { Flags } from '@oclif/core';
+import * as fs from 'fs-extra';
+
+import { withSudoModeAsync } from '../../authUtils';
+import EasCommand from '../../commandUtils/EasCommand';
+import { EASEnvironmentFlag, EASNonInteractiveFlag } from '../../commandUtils/flags';
+import {
+  EnvironmentVariableEnvironment,
+  EnvironmentVariableFragment,
+} from '../../graphql/generated';
+import { EnvironmentVariablesQuery } from '../../graphql/queries/EnvironmentVariablesQuery';
+import Log from '../../log';
+import { confirmAsync } from '../../prompts';
+
+type PullFlags = {
+  environment?: EnvironmentVariableEnvironment;
+  path: string;
+  'non-interactive': boolean;
+};
+
+export default class EnvironmentValuePull extends EasCommand {
+  static override description = 'pull env file';
+
+  static override hidden = true;
+
+  static override contextDefinition = {
+    ...this.ContextOptions.ProjectConfig,
+    ...this.ContextOptions.LoggedIn,
+    ...this.ContextOptions.SessionManagment,
+  };
+
+  static override flags = {
+    ...EASEnvironmentFlag,
+    ...EASNonInteractiveFlag,
+    path: Flags.string({
+      description: 'Path to the result `.env` file',
+      default: '.env.eas.local',
+    }),
+  };
+
+  async runAsync(): Promise<void> {
+    const { flags } = await this.parse(EnvironmentValuePull);
+    let {
+      environment,
+      path: targetPath,
+      'non-interactive': nonInteractive,
+    } = this.validateFlags(flags);
+
+    const {
+      privateProjectConfig: { projectId },
+      loggedIn: { graphqlClient },
+      sessionManager,
+    } = await this.getContextAsync(EnvironmentValuePull, {
+      nonInteractive,
+    });
+
+    if (!nonInteractive) {
+      const confirm = await confirmAsync({
+        message: `Pull the environment variables for the ${environment} environment to a local ${targetPath} file. Do you want to continue?`,
+      });
+      if (!confirm) {
+        Log.log('Aborting...');
+        return;
+      }
+    }
```

Comment:
If it's excessive, I believe this is user's intention when running this command

Reply:
Now that we pull it to a separate file I agree, but if we were to overwrite existing `.env.local` file it would be good to confirm this.